File: Bind75/1207/Solution.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function(arr) {
    let flag =true;
    // step 1
    let cntMap = new Map();
    for(let i=0;i<arr.length;i++){
        k = arr[i];
        v=1;
        if(cntMap.has(k)){
            v= cntMap.get(k) +1;
        }
        cntMap.set(k,v);
    }

    // step 2 :
    let set = new Set(cntMap.values());
    if(set.size < cntMap.size){
        flag=false;
    }
    return flag;
};

File: Bind75/1207/Solution.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        boolean flag = true;
        // step 1
        HashMap<Integer,Integer> cntMap = new HashMap<>();
        for(int num:arr){
            int v=1;
            if(cntMap.containsKey(num)){
                v = cntMap.get(num)+1;
            }
            cntMap.put(num, v);
        }

        // step 2:
        Set<Integer> set = new HashSet<>(cntMap.values());
        if(set.size()< cntMap.size()){flag=false;}
        return flag;
    }
}